mlir/include/mlir/Analysis/Presburger/Matrix.h

```diff
@@ -239,6 +236,36 @@ class IntMatrix : public Matrix<MPInt> {
   /// Divide the columns of the specified row by their GCD.
   /// Returns the GCD of the columns of the specified row.
   MPInt normalizeRow(unsigned row);
+
+  // Return the integer inverse of the matrix, leaving the calling object
+  // unmodified.
+  std::optional<IntMatrix> integerInverse();
+
+  // Compute the determinant of the matrix by converting it to row echelon
+  // form and then taking the product of the diagonal.
+  MPInt determinant();
+};
+
+// An inherited class for rational matrices, with no new data attributes.
+// This class is for functionality that only applies to matrices of fractions.
+class FracMatrix : public Matrix<Fraction> {
+public:
+  FracMatrix(unsigned rows, unsigned columns, unsigned reservedRows = 0,
+             unsigned reservedColumns = 0)
+      : Matrix<Fraction>(rows, columns, reservedRows, reservedColumns){};
+
+  FracMatrix(Matrix<Fraction> m)
+      : Matrix<Fraction>(m) {};
+
+  /// Return the identity matrix of the specified dimension.
+  static FracMatrix identity(unsigned dimension);
+
+  // Return the inverse of the matrix, leaving the calling object unmodified.
+  std::optional<FracMatrix> inverse();
```

Comment:
if the calling object is unmodified then add the const keyword

`std::optional<FracMatrix> inverse() const;`